openedx_rocketchat/api/v0/views.py

```diff
@@ -0,0 +1,255 @@
+
+            auth_token = data.get('authToken', None)
+            user_id = data.get('userId', None)
+
+            serializer = RocketChatCredentialsSerializer(
+                data={
+                    "url_service": url_service,
+                    "auth_token": auth_token,
+                    "user_id": user_id,
+                }
+            )
+            if serializer.is_valid():
+                return Response(serializer.data, status=status.HTTP_200_OK)
+
+        LOG.error("Rocketchat API object can not be initialized")
+        return Response(status=status.HTTP_503_SERVICE_UNAVAILABLE)
+
+
+class RocketChatSubscriptionsId(APIView):
+    """
+    Class that allows to get the id for every subscription.
+    """
+
+    authentication_classes = (
+        SessionAuthentication,
+    )
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def get(self, request):  # pylint: disable=no-self-use
+        """
+        Retrieve the rooms' id for every user subscription.
+
+        **Example Requests**:
+
+            GET /rocketchat/api/v0/subscriptions-id?unread=true&authToken=asdjfsdhfgasdas&&userId=86435f4asd56f4a
+
+            * unread : True to retrieve only subscription where the user has unread mentions.userId
+            * authToken : Rocketchat authorization token.
+            * userId : The unique rocketchat identifier for the user.
+
+        **Response Values**:
+
+            * subscriptions_id : A list with the room's id for every user subscription.
+
+        """
+        try:
+            unread = request.GET.get("unread")
+            auth_token = request.GET.get("authToken")
+            user_id = request.GET.get("userId")
+            serializer = RocketChatSubscriptionsIdSerializer(
+                data={
+                    "subscriptions_id": list(get_subscriptions_rids(auth_token, user_id, unread))
+                }
+            )
+            if serializer.is_valid():
+
+                return Response(serializer.data, status=status.HTTP_200_OK)
+
+        except Exception as error:
+            LOG.error("Rocketchat Subscriptions couldn't be got due to %s", error)
+
+        return Response(status=status.HTTP_400_BAD_REQUEST)
+
+
+class RocketChatChangeRole(APIView):
+    """
+    Class that allows to change the role for a given user.
+    """
+
+    authentication_classes = (
+        SessionAuthentication,
+    )
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def post(self, request):  # pylint: disable=no-self-use
+        """
+        This methods allows to change the rocketchat role for a specific user
+
+        **Example Requests**:
+
+            POST /rocketchat/api/v0/change-role
+
+            * data = {
+                "username": "edx",
+                "role": "admin",
+            }
+
+        **Response Values**:
+
+            * It doesn't return data on successful.
+            * {"error": "error reason"} on fail by  404.
```

Comment:
Please, fix the status response code. @andrey-canon

Reply:
ok